tasks/main.yml

```diff
@@ -17,26 +12,328 @@
-  service:
-    name: "{{ item }}"
-    state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
-
-- name: Generate /etc/{{ __template_foo_config }}
-  template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: yes
-    mode: 0400
-  notify: template handler to restart services
+- name: Check if the errorlogfile setting has a custom value
+  command: /opt/mssql/bin/mssql-conf get filelocation errorlogfile
+  register: __mssql_errorlogfile
+  ignore_errors: true
+  changed_when: false
+
+- name: Check if the errorlog file exists in a custom location
+  stat:
+    path: "{{ __mssql_errorlogfile.stdout | regex_search('[^: ]*$') }}"
+  register: __mssql_errorlog_custom
+  when: "'errorlogfile' in __mssql_errorlogfile.stdout"
+
+- name: Check if the errorlog file exists in the default location
+  stat:
+    path: /var/opt/mssql/log/errorlog
+  register: __mssql_errorlog_default
+  when: "'errorlogfile' not in __mssql_errorlogfile.stdout"
+
+- name: Gather system services facts
+  service_facts:
+
+- name: Verify that the variables required for setting up MSSQL are defined
+  assert:
+    that:
+      - mssql_password is defined
+      - mssql_edition is defined
+    fail_msg:
+      - "You must define the following variables to set up MSSQL Server:"
+      - "mssql_password"
+      - "mssql_edition"
+  when: >
+    ("running" not in
+    ansible_facts["services"]["mssql-server.service"]["state"]) and
+    ("enabled" not in
+    ansible_facts["services"]["mssql-server.service"]["status"]) and
+    (((__mssql_errorlog_custom.stat is defined) and
+    (not __mssql_errorlog_custom.stat.exists)) or
+    ((__mssql_errorlog_default.stat is defined) and
+    (not __mssql_errorlog_default.stat.exists)))
+  register: __mssql_variables_evaluation
+
+- name: Set up MS SQL Server
+  command: /opt/mssql/bin/mssql-conf -n setup
+  environment:
+    - ACCEPT_EULA: "Y"
+    - MSSQL_SA_PASSWORD: "{{ mssql_password }}"
+    - MSSQL_PID: "{{ mssql_edition }}"
+  when: >
+    ("running" not in
+    ansible_facts["services"]["mssql-server.service"]["state"]) and
+    ("enabled" not in
+    ansible_facts["services"]["mssql-server.service"]["status"]) and
+    (((__mssql_errorlog_custom.stat is defined) and
+    (not __mssql_errorlog_custom.stat.exists)) or
+    ((__mssql_errorlog_default.stat is defined) and
+    (not __mssql_errorlog_default.stat.exists)))
+  register: __mssql_conf_setup
+
+- name: Configure the Microsoft SQL Server Tools repository
+  yum_repository:
+    name: packages-microsoft-com-prod
+    description: Microsoft SQL Server Tools
+    baseurl: "{{ __mssql_client_repository }}"
+    gpgcheck: true
+
+- name: Ensure that SQL Server client tools are installed
+  package:
+    name: "{{ __mssql_client_packages }}"
+    state: present
+  environment:
+    - ACCEPT_EULA: Y
+
+- name: Set a new password for the MSSQL Server sa user
+  when:
+    - __mssql_conf_setup.skipped is defined
```

Comment:
use `__mssql_conf_setup is skipped` - see https://docs.ansible.com/ansible/2.8/user_guide/playbooks_tests.html#task-results